rclcpp/src/rclcpp/serialized_message.cpp

```diff
@@ -66,7 +66,7 @@ SerializedMessage::SerializedMessage(const rcl_serialized_message_t & other)
 }
 
 SerializedMessage::SerializedMessage(SerializedMessage && other)
-: SerializedMessage(other.serialized_message_)
+: SerializedMessage(static_cast<rcl_serialized_message_t &&>(other.serialized_message_))
```

Comment:
So this is correct, and I can't think of a reason why it wouldn't work, but I think that this would be more obvious:

```
SerializedMessage::SerializedMessage(SerializedMessage && other)
: serialized_message_(rmw_get_zero_initialized_serialized_message())
{
  std::swap(this->serialized_message_, other->serialized_message_);
}
```

Reply:
fair enough. I'll take it. :)

Reply:
Or if you want the one-liner:

```
SerializedMessage::SerializedMessage(SerializedMessage && other)
: serialized_message_(std::exchange(other.serialized_message_, rmw_get_zero_initialized_serialized_message()))
{
}
```

Reply:
addressed in https://github.com/ros2/rclcpp/pull/1097/commits/205ab58db922ba530f0e6f096a0c7be8e383a47c